Feature/multi context execution

jdzrtz opened the pull request:
Multicontext command execution.

Commit:
feat(): step not found does not stop the flow

Commit:
feat(): multicontext execution

Commit:
refactor(): rename syncNormals

Commit:
fix(): right context in analytics

Commit:
feat(): check if the command has a context of execution

Commit:
feat(): multicontext detection changes

- getFlowParams returns now an object containing flow params for each context

- getStepParams returns now an object containing step params for each context

- getContexts now considers a flow is implemented by a context if any of its steps is implemented for that context

Commit:
feat(): multicontext

- setFlow: includes a new level for each of the contexts of execution

Commit:
feat(): multicontext

- syncContexts: removed, the context by context execution is now performed in stepper

- _addOutputs: outputs and inputs for any step are mapped by context

Commit:
feat(): multicontext reporting

- report: add context as additional report level

Commit:
feat(): multicontext

- syncContexts: now the stepper executes the step for each of the contexts in the normal

Commit:
chore(): show flow name

Commit:
fix(): multicontext

- include context in step reporting

Commit:
feat(): show individual context checked in message

Commit:
style(): code format issues

Commit:
style(): code linting issues

Commit:
chore(): stepper errors colors

froucher commented:
Maybe, it would be better to merge to the branch 'develop', because before finish this release, others feature have to be merged too (like change the interactive list of flow and steps).

Commit:
feat(): reporter module, initial commit

Commit:
feat(): Step.report uses reporter module

Commit:
refactor(): remove deprecated functions "save" and "get"

Commit:
feat(): add "warnings" field

Commit:
feat(): step reporting functions

- readReportsByName
- createStepReport
- ensureStepReport
- addReportResult

Commit:
feat(): report step not implemented as warning

Commit:
feat(): use reporter module

Commit:
feat(): return hardOk value at finish

Commit:
feat(): state improvements

- use reporter module to access reports
- return a complex result which calculates if is a hardOk or a softOk

Commit:
feat(): improves subflows execution

- if a subflow finishes as a softOk the execution continues

Commit:
fix(): show correct flow name
